### ISSUE TITLE
Unexpected comma with valid Ruby code that uses splats for asignment

### ISSUE THREAD

**Issue by RickCarlino** [opal/opal#881]
This is kind of an edge case / uncommon Ruby feature, but I was able to break the compiler with the following code:

[Live sample](http://opalrb.org/try/?code:key%20%3D%20%22do_not_include_this%22%0A*output%2C%20_discard%20%3D%20key.split(%27_%27)%0A%0Aif%20output.include%3F(%22this%22)%0A%20%20puts%20%27fail%27%0Aelse%0A%20%20puts%20%27expected%20behavior%27%0Aend%0A%0A)

```ruby
key = "do_not_include_this"
*output, _discard = key.split('_')

if output.include?("this")
  puts 'fail'
else
  puts 'expected behavior'
end
```

Results in:

```
unexpected '"tCOMMA"'
Source: (file):2:8
*output, _discard = key.split('_')
~~~~~~~^
SyntaxError: An error occurred while compiling: (file)

unexpected '"tCOMMA"'
Source: (file):2:8
*output, _discard = key.split('_')
~~~~~~~^
    at t.$new (http://opalrb.org/javascripts/try.js:5:30463)
    at a.<anonymous> (http://opalrb.org/javascripts/try.js:5:26369)
    at a.d.$compile (http://opalrb.org/javascripts/try.js:19:4138)
    at e.$compile (http://opalrb.org/javascripts/try.js:19:2017)
    at $TryOpal.def.$run_code (http://opalrb.org/javascripts/try.js:20:27039)
    at HTMLDocument.$a.$$p.TMP_3 (http://opalrb.org/javascripts/try.js:20:28057)
    at f (http://opalrb.org/javascripts/application.js:2:16766)
    at Object.p.fireWith [as resolveWith] (http://opalrb.org/javascripts/application.js:2:17547)
    at Function.K.extend.ready (http://opalrb.org/javascripts/application.js:2:12308)
    at HTMLDocument.pt (http://opalrb.org/javascripts/application.js:2:9736)
```

**Comment by alexdowad** [comment]
I just pushed a `rubyspec_variables` branch which makes Opal's mass assignment fully Rubyspec-compliant. This issue will be fixed as a result. Just seeing if it passes CI...

**Comment by alexdowad** [comment]
Everything is green, merging. This problem is now fixed.

**Comment by elia** [comment]
(fixed by 0f6e09f8cad575fd802ff5b20174b2f250afa7e0)
